stock_picking_auto_create_lot_qty/models/stock_picking.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
+        """
+        Allows to be called either by button or through code
+        """
+        pickings = self.filtered(lambda p: p.picking_type_id.auto_create_lot)
+        lines = pickings.mapped("move_line_ids").filtered(
+            lambda x: (
+                not x.lot_id
+                and not x.lot_name
+                and x.product_id.tracking != "none"
+                and x.product_id.auto_create_lot
+            )
+        )
+        new_lines = pickings.mapped("move_line_ids").filtered(lambda x: (x.qty_done))
+        if not new_lines:
+            count_by_product = {}
+            for line in lines:
+                if line.product_id not in count_by_product:
+                    count_by_product[line.product_id] = line.product_uom_qty
+                else:
+                    count_by_product[line.product_id] += line.product_uom_qty
+            for product_id, product_qty in count_by_product.items():
+                current_product_qty = product_id.uom_id._compute_quantity(
+                    product_qty,
+                    product_id.batch_uom_id or product_id.uom_id,
+                    round=False,
+                )
+                if product_id.create_lot_every_n and product_id.only_multiples_allowed:
+                    raise UserError(
+                        _(
+                            "The quantity received for the product does"
+                            " not allow for automatic lot creation"
+                        )
+                    )
+                if (
+                    product_id.create_lot_every_n
+                    and not product_id.only_multiples_allowed
+                ):
+                    every_n = product_id.product_tmpl_id.create_lot_every_n
+
+                    count_of_lots = int(current_product_qty // every_n)
```

Comment:
```suggestion
                    count_of_lots = current_product_qty // every_n
```

Floor division already returns an integer

Reply:
`    for _new_line_count in range(count_of_lots):
TypeError: 'float' object cannot be interpreted as an integer`